xbmc/filesystem/posix/PosixFile.cpp

```diff
@@ -313,6 +314,18 @@ bool CPosixFile::Rename(const CURL& url, const CURL& urlnew)
   
   if (rename(name.c_str(), newName.c_str()) == 0)
     return true;
+
+  // rename across mount points - need to copy/delete
+  if (errno == EXDEV)
+  {
+    if (XFILE::CFile::Copy(name, newName))
+    {
+      if (XFILE::CFile::Delete(name))
```

Comment:
Code/scripts mostly depend on target file presence.  If full copy is present it's better to let script/code to continue working.
Don't think that anyone will handle presence of source file after rename as special case.